tests/phpunit/integration/Core/Util/Google_URL_Matcher_TraitTest.php

```diff
@@ -205,6 +208,106 @@ public function data_is_domain_match() {
 				'turkish.com',
 				false,
 			),
+			'domain with and without bidirectional control characters' => array(
+				'xn--5dbedds0a1eb.xn--9dbq2a',
+				'‬ביקדקוםה.קום',
+				true,
+			),
+		);
+	}
+
+	/**
+	 * @dataProvider data_url_normalizer
+	 *
+	 * @param string $domain   A domain.
+	 * @param string $compare  The domain or URL to compare.
+	 * @param bool   $expected Whether a match is expected or not.
+	 */
+	public function test_url_normalizer( $domain, $compare, $expected ) {
```

Comment:
This entire `Google_URL_Normalizer`-specific test code should go into a separate test class, there should always be one test class per class to test. In other words this should go into `Google\Site_Kit\Tests\Core\Util\Google_URL_NormalizerTest`.